### DISCUSSION TITLE
API Questions: Documentation and Functionality

### DISCUSSION THREAD

**Discussion by Cadair** [magicbug/Cloudlog#1484] [Ideas / Feature Requests]
Hello,

I have been toying with the idea of building a few things which make use of my cloudlog instance, starting from simple and going all the way up to a whole new frontend.

In the process of thinking about this I have been looking into the API that cloudlog provides, and I can't seem to find much documentation on it? From my scanning of the code it seems that a lot of the functionality the frontend uses isn't part of the "API" (at least as defined in `Api.php`) but in various other places? Is this correct?

I am a Python developer, and I have recently started using [FastAPI](https://fastapi.tiangolo.com/) and one of the *amazing* things that it provides is automatically is an OpenAPI / Swagger API documentation page. I was looking into generating one of these in php and found things like https://github.com/zircote/swagger-php

I was wondering what you would think about a PR adding annotations and API docs to what's in `Api.php` as a starter, and then perhaps moving onto adding more things to the "public" API?

Thanks!

**Comment by magicbug** [comment]
Fully support any improvement to the API and agree the docs for it suck its just this page https://github.com/magicbug/Cloudlog/wiki/API  or you have to read the code. If you fancy trying to improve things go for it
